commit d325c162631eb7c21f244f48184f3c8ace868039
Author: Akinobu Mita <akinobu.mita@gmail.com>
Date:   Sat Apr 15 21:57:05 2023 +0900

    fault-inject: fix build error when FAULT_INJECTION_CONFIGFS=y and CONFIGFS_FS=m
    
    This fixes a build error when CONFIG_FAULT_INJECTION_CONFIGFS=y and
    CONFIG_CONFIGFS_FS=m.
    
    Since the fault-injection library cannot built as a module, avoid building
    configfs as a module.
    
    Fixes: 4668c7a2940d ("fault-inject: allow configuration via configfs")
    Reported-by: kernel test robot <lkp@intel.com>
    Link: https://lore.kernel.org/oe-kbuild-all/202304150025.K0hczLR4-lkp@intel.com/
    Signed-off-by: Akinobu Mita <akinobu.mita@gmail.com>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index e700b29d7756..48375e027b09 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1960,7 +1960,8 @@ config FAIL_SUNRPC
 
 config FAULT_INJECTION_CONFIGFS
 	bool "Configfs interface for fault-injection capabilities"
-	depends on FAULT_INJECTION && CONFIGFS_FS
+	depends on FAULT_INJECTION
+	select CONFIGFS_FS
 	help
 	  This option allows configfs-based drivers to dynamically configure
 	  fault-injection via configfs.  Each parameter for driver-specific